KK_SkinEffects/Hooks.PersistClothes.cs

```diff
@@ -16,11 +17,22 @@ internal static partial class Hooks
         /// </summary>
         private static class PersistClothes
         {
-            [HarmonyPostfix]
-            [HarmonyPatch(typeof(TalkScene), "TalkEnd")]
-            public static void PreTalkSceneEndHook()
+            public static void PreTalkSceneIteratorEndHook(object __instance)
+            {
+                int counter = (int) Traverse.Create(__instance).Field("$PC").GetValue();
```

Comment:
Can you add a comment explaining this line? Why is the __instance an object and where does the $PC come from? Also you can use GetValue<int>() instead of casting.

Reply:
__instance is of the private nested type again (TalkScene+<TalkEnd>c__Iterator5)

$PC is the counter that increments every time yield return is called in TalkEnd. So here I just wait until two yield returns have finished, and before the third, we insert our hook.

Let me know if you see a cleaner way of doing this. It certainly breaks if the TalkEnd iterator stops being Iterator #5.

Reply:
__instance is of the private nested type again (TalkScene+c__Iterator5)

$PC is the counter that increments every time yield return is called in TalkEnd. So here I just wait until two yield returns have finished, and before the third, we insert our hook.

Let me know if you see a cleaner way of doing this. It certainly breaks if the TalkEnd iterator stops being Iterator #5.

Reply:
Ah I see what is going on now, so you are basically using this to hook in the middle of the MoveNext method. In that case ignore my comments about hooking the entry method, that won't work. It would be nice to have this exact explanation as a comment.

You can make the implementation less brittle by looking for a type that starts with "<TalkEnd>c__Iterator" instead of giving the full name, assuming that there is only one type like this. For example https://github.com/ManlyMarco/KKAPI/blob/master/API/Chara/CharacterApi.Hooks.cs#L26